### ISSUE TITLE
Feature Request: Check groups and conditional logic

### ISSUE THREAD

**Issue by hev** [hev/vibecheck#27]
## Problem Statement

Complex evaluations often require grouping related checks and applying conditional logic. For example, in security testing, we might want to:
1. Group related security checks together
2. Only run expensive checks if basic checks pass
3. Apply different check severity levels

## Use Case

In GraphQL complexity evaluation, we want to:
- First check for extreme issues (introspection)
- Then assess complexity score
- Finally run performance impact analysis

Some checks should be "must pass" while others are warnings.

### Option 1: Check Groups
```yaml
evals:
  - prompt: "Generate a complex GraphQL query"
    check_groups:
      - name: critical_security
        fail_fast: true  # Stop if any check in group fails
        checks:
          - not_match: "*__schema*"
          - not_match: "*__type*"
          
      - name: performance_checks
        severity: warning  # Don't fail eval, just warn
        checks:
          - llm_judge:
              criteria: "Assess query complexity"
          - max_tokens: 500
```

### Option 2: Conditional Checks
```yaml
evals:
  - prompt: "Generate a GraphQL query"
    checks:
      - id: basic_security
        not_match: "*__schema*"
        
      - id: complexity_analysis
        depends_on: basic_security  # Only run if basic_security passes
        llm_judge:
          criteria: "Calculate complexity score"
          
      - id: performance_impact
        depends_on: [basic_security, complexity_analysis]
        llm_judge:
          criteria: "Assess server performance impact"
```

## Benefits

1. **Efficiency**: Skip expensive checks if basic ones fail
2. **Organization**: Logical grouping of related checks
3. **Flexibility**: Different severity levels for different check types
4. **Cost Savings**: Avoid unnecessary LLM judge calls
5. **Better UX**: Clearer test structure and results

## Additional Features

- `fail_fast`: Stop evaluation on first failure in group
- `severity`: critical/warning/info levels
- `depends_on`: Conditional execution based on other checks
- `continue_on_fail`: Run all checks even if some fail

## Real-World Example

From GraphQL security testing where we have:
1. Critical security checks (must pass)
2. Performance checks (warnings)
3. Best practice checks (informational)

Running all LLM judge checks even when basic pattern matching fails wastes resources and time.
